clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -269,6 +301,9 @@ protected Struct toStruct() {
             struct.set(REBALANCE_TIMEOUT_KEY_NAME, rebalanceTimeout);
         }
         struct.set(MEMBER_ID, memberId);
+        if (version >= 5) {
+            struct.set(GROUP_INSTANCE_ID, groupInstanceId);
```

Comment:
So clearly the intent is to silently fall back to the old join group logic, which means we become a dynamic member. It may be helpful having a log message indicating that this has happened.

One additional note: if the brokers are later upgraded to a version that does support static membership, we don't have any logic to detect it. I think this is probably fine, just worth keeping in mind.